public/app/features/dashboard/components/DashNav/DashNav.tsx

```diff
@@ -102,6 +113,60 @@ class DashNav extends PureComponent<Props> {
     this.forceUpdate();
   };
 
+  addCustomContent(actions: DashNavButtonModel[], buttons: ReactNode[]) {
+    actions.map((action, index) => {
+      const Component = action.component;
+      const element = <Component {...this.props} key={`button-custom-${index}`} />;
+      typeof action.index === 'number' ? buttons.splice(action.index, 0) : buttons.push(element);
```

Comment:
Should be `buttons.splice(action.index, 0, element)`, seems - otherwise element won't be inserted.
```suggestion
      typeof action.index === 'number' ? buttons.splice(action.index, 0, element) : buttons.push(element);
```